src/modules/PowerShell/FinOpsHub/FinOpsHub.psm1

```diff
@@ -0,0 +1,193 @@
+    param()
+    $context = Get-AzContext
+    if ($null -eq $context)
+    {
+        Write-Warning -Message ($LocalizedData.NotLoggedIn)
+        $null = Login-AzAccount
+        $context = Get-AzContext
+    }
+
+    Write-Verbose -Message ($LocalizedData.LoggedIn -f $context.Subscription.Name, $context.Subscription.Id)
+    return $context
+}
+
+<#
+    .SYNOPSIS
+        Deploys a FinOpsHub instance.
+
+    .PARAMETER Path
+        Path to the .bicep template.
+
+    .PARAMETER HubName
+        Name of the FinOpsHub instance.
+
+    .PARAMETER Scope
+        Scope to deploy FinOpsHub to, either ResourceGroup, Subscription, or Tenant.
+
+    .PARAMETER ResourceGroupName
+        Name of the resource group to deploy to. Will be created if it doesn't exist. Only used for ResourceGroup scope.
+
+    .PARAMETER Location
+        Azure location to execute the deployment from.
+
+    .PARAMETER StorageSku
+        Optional. Storage account SKU. LRS = Lowest cost, ZRS = High availability. Note Standard SKUs are not available for Data Lake gen2 storage.
+
+    .PARAMETER Tags
+        Optional. Tags for all resources.
+
+    .PARAMETER ExportScopes
+        Optional. List of scope IDs to create exports for.
+
+    .EXAMPLE
+        Deploy-FinOpsHub -Path c:\finopshub\main.bicep -HubName MyHub -ResourceGroupName MyExistingResourceGroup -Location westus -Scope ResourceGroup
+        
+        Deploys a new FinOps hub instance named MyHub to an existing resource group name MyExistingResourceGroup.
+    
+    .EXAMPLE
+        Deploy-FinOpsHub -Path c:\finopshub\main.bicep -HubName MyHub -Location westus -Scope Subscription
+        
+        Deploys a new FinOps hub instance named MyHub to your currently logged into Azure subscription.
+
+    .EXAMPLE
+        Deploy-FinOpsHub -Path c:\finopshub\main.bicep -HubName MyHub -Location westus -Scope Tenant
+        
+        Deploys a new FinOps hub instance named MyHub to your currently logged into Azure tenant.
+#>
+function Deploy-FinOpsHub
+{
+    [CmdletBinding()]
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [ValidateScript({Test-Path -Path $_})]
+        [ValidateScript({{[System.IO.Path]::GetExtension($_) -eq '.bicep'}})]
+        [string]
+        $Path,
+
+        [Parameter(Mandatory = $true)]
+        [string]
+        $HubName,
+
+        [Parameter(Mandatory = $true)]
+        [string]
+        $Location,
+
+        [Parameter()]
+        [ValidateSet('ResourceGroup', 'Subscription', 'Tenant')]
+        [string]
+        $Scope = 'ResourceGroup',
+
+        [Parameter()]
+        [ValidateSet('Premium_LRS', 'Premium_ZRS')]
+        [string]
+        $StorageSku,
+
+        [Parameter()]
+        [hashtable]
+        $Tags,
+
+        [Parameter()]
+        [array]
+        $ExportScopes
+    )
+
+    DynamicParam
+    {
+        if ($Scope -eq 'ResourceGroup')
+        {
+            $attribute = New-Object System.Management.Automation.ParameterAttribute
```

Comment:
nit: Why not use `@{ ... }`? I def prefer the simpler syntax for something as basic as a hashtable.

Reply:
This is for a dynamic parameter, though I'm not sure it's necessary anymore if resource group will always be the target for the hub.

Reply:
not needed anymore